src/internal/analytics-metadata/__tests__/labels-utils.test.tsx

```diff
@@ -0,0 +1,228 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { getLabelFromElement, processLabel } from '../labels-utils';
+import { getAnalyticsMetadataAttribute, getAnalyticslabelAttribute } from '../attributes';
+
+describe('getLabelFromElement', () => {
+  test('returns an empty string if the element is null', () => {
+    expect(getLabelFromElement(null)).toEqual('');
+  });
+  test('returns text content', () => {
+    const { container } = render(
+      <div>
+        <div id="target1">content</div>
+        <div id="target2">
+          <span>text 1</span>
+          <span>text 2</span>
+        </div>
+      </div>
+    );
+    const target1 = container.querySelector('#target1');
+    expect(getLabelFromElement(target1 as HTMLElement)).toEqual('content');
+    const target2 = container.querySelector('#target2');
+    expect(getLabelFromElement(target2 as HTMLElement)).toEqual('text 1text 2');
+  });
+  test('returns aria-label if defined', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-label="returned label" aria-labelledby="id:r55:">
+          content
+        </div>
+        <div id="id:r55:" aria-label="labelled by label">
+          content
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('returned label');
+  });
+  test('returns aria label of aria-labelledby element', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55:">
+          content
+        </div>
+        <div id="id:r55:" aria-label="labelled by label">
+          content
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('labelled by label');
+  });
+  test('returns text content of aria-labelledby element', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55:">
+          content
+        </div>
+        <div id="id:r55:">second content</div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('second content');
+  });
+  test('returns text content of first aria-labelledby element', () => {
+    const { container } = render(
+      <div>
+        <div id="target" aria-labelledby="id:r55: id:r56:">
+          content
+        </div>
+        <div id="id:r55:">second content</div>
+        <div id="id:r56:">third content</div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('second content');
+  });
+  test('returns empty string if aria-labelledby element does not exist', () => {
```

Comment:
Would textContent be better here?

Reply:
I think that if there is an aria-labelledby that points to nothing, there is an issue. I prefer to return an empty string